### PR TITLE
Added Cli Option to include external dependencies 

### LINKED ISSUE #860: Setting to apply external dependencies in Non-interactive mode
Today's behaviour defaults to not applying external dependencies in non-interactive mode as we can see in [these lines of code](https://github.com/gruntwork-io/terragrunt/blob/8fa2f715589fcba6c6e0131e4d954b9b4264463d/configstack/module.go#L436).
I would like to have some configuration (for example a cli option) to apply the external dependencies even though the non-interactive mode has been set on.

### PR DESCRIPTION
**Pull request by gcrespi**
on *-all commands without asking on the prompt.
This PR fixes #860

### CODE REVIEW

**Comment by brikis98** [comment]
Sanity check: is there a way to accomplish each of the following?

1. `terragrunt apply-all` that automatically includes all external dependencies.
1. `terragrunt apply-all` that automatically excludes all external dependencies.
1. `terragrunt destroy-all` that automatically includes all external dependencies.
1. `terragrunt destroy-all` that automatically excludes all external dependencies.

We should probably have tests for all of these too!

**Comment by gcrespi** [comment]
> Sanity check: is there a way to accomplish each of the following?
> 
> 1. `terragrunt apply-all` that automatically includes all external dependencies.
> 2. `terragrunt apply-all` that automatically excludes all external dependencies.
> 3. `terragrunt destroy-all` that automatically includes all external dependencies.
> 4. `terragrunt destroy-all` that automatically excludes all external dependencies.
> 
> We should probably have tests for all of these too!

1. `terragrunt apply-all --terragrunt-auto-apply-external-dependencies`
2. `terragrunt apply-all --terragrunt-ignore-external-dependencies`
3. `terragrunt destroy-all --terragrunt-auto-apply-external-dependencies`
4. `terragrunt destroy-all --terragrunt-ignore-external-dependencies`

I've added some tests recently:
- one to test that the flag actually sets the new option variable.
- the other was an allready existing integration test that was ignored due to the lack of possibility to apply external dependencies o non interactive mode until now.

**Comment by brikis98** [comment]
Thx @gcrespi. After seeing this:

```
terragrunt destroy-all --terragrunt-auto-apply-external-dependencies
terragrunt destroy-all --terragrunt-ignore-external-dependencies
```

Perhaps `--terragrunt-auto-apply-external-dependencies` should be called `--terragrunt-include-external-dependencies`?

**Comment by gcrespi** [comment]
Ok @brikis98, sounds good to me. I've just renamed it.

File: configstack/module.go [hunk 1]
```diff
@@ -434,6 +434,11 @@ func resolveExternalDependenciesForModule(module *TerraformModule, moduleMap map
 // Confirm with the user whether they want Terragrunt to assume the given dependency of the given module is already
 // applied. If the user selects "yes", then Terragrunt will apply that module as well.
 func confirmShouldApplyExternalDependency(module *TerraformModule, dependency *TerraformModule, terragruntOptions *options.TerragruntOptions) (bool, error) {
+	if terragruntOptions.includeExternalDependencies {
+		terragruntOptions.Logger.Printf("The --terragrunt-include-external-dependencies flag is set. So external dependencies with an xxx-all command, will run this command against module %s, which is a dependency of module %s.", dependency.Path, module.Path)
```
Comment (brikis98) [inline configstack/module.go]:
```suggestion
		terragruntOptions.Logger.Printf("The --terragrunt-include-external-dependencies flag is set, so automatically including all external dependencies, and will run this command against module %s, which is a dependency of module %s.", dependency.Path, module.Path)
```

**Comment by brikis98** [comment]
Build failed:

```
# github.com/gruntwork-io/terragrunt/configstack
configstack/module.go:437:22: terragruntOptions.includeExternalDependencies undefined (cannot refer to unexported field or method includeExternalDependencies)
# github.com/gruntwork-io/terragrunt/configstack [github.com/gruntwork-io/terragrunt/configstack.test]
configstack/module.go:437:22: terragruntOptions.includeExternalDependencies undefined (cannot refer to unexported field or method includeExternalDependencies)
```

**Comment by gcrespi** [comment]
@brikis98 
> configstack/module.go:437:22: terragruntOptions.includeExternalDependencies undefined (cannot refer to unexported field or method includeExternalDependencies)
Sorry, there was a typo there, the variable initial letter was lowercase :sweat_smile:

File: cli/args.go [hunk 2]
```diff
@@ -85,6 +85,8 @@ func parseTerragruntOptionsFromArgs(args []string, writer, errWriter io.Writer)
 
 	ignoreExternalDependencies := parseBooleanArg(args, OPT_TERRAGRUNT_IGNORE_EXTERNAL_DEPENDENCIES, false)
 
+	IncludeExternalDependencies := parseBooleanArg(args, OPT_TERRAGRUNT_INCLUDE_EXTERNAL_DEPENDENCIES, false)
```
Comment (brikis98) [inline cli/args.go]: Should be lower case by Go's naming conventions. Upper case is only used for public fields (e.g., on a struct, as the field name on `terragrunt.Options`).

Comment (gcrespi) [reply cli/args.go]: Fixed that, sorry!

Comment (brikis98) [reply cli/args.go]: Thx!

**Comment by brikis98** [comment]
Kicking off tests again

**Comment by brikis98** [comment]
OK, test passed! Merging now.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.19.27

**Pull request merged**